tests/test_sampler.py

```diff
@@ -4,32 +4,87 @@
 
 from common import get_context
 
+def checkerror(func):
```

Comment:
I think we need no checks on gl errors. The initialization step results in gl errors for some gl get calls. I don't want to clear them. I also implemented a standalone context that uses the `GL_CONTEXT_FLAG_NO_ERROR_BIT`.
Sometimes I install hooks on the opengl calls to check for gl errors (unfortunately these calls will ensure error=0).
I think tests should not depend on opengl error values since they are changed when inspected.

Reply:
Fair enough.